File: 1.两数之和.cpp
/*
 * @lc app=leetcode.cn id=1 lang=cpp
 *
 * [1] 两数之和
 */
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int length = nums.size();
        unordered_map<int,int> m;
        vector<int> result;
        for(int i=0;i<length;i++)
        {
            int a = nums[i];
            if(m.find(target-a)!=m.end())
            {
                result.push_back(i);
                result.push_back(m[target-a]);
            }
            m[a] = i;
        }
        return result;
    }
};

File: 1.两数之和.java
// import java.util.HashMap;

/*
 * @lc app=leetcode.cn id=1 lang=java
 *
 * [1] 两数之和
 */
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int result[] = new int[2];
        HashMap<Integer,Integer> m = new HashMap<>();
        int length = nums.length;
        for(int i=0;i<length;i++)
        {
            int a = nums[i];
            if(m.containsKey(target-a))
            {
                // result =  new int[]{i,m.get((target-a))};
                result[0] = i;
                result[1] = m.get((target-a));
            }
            m.put(nums[i], i);   
        }
        return result;
    }
}

